config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  get 'posts/index'
+  get 'posts/show'
+  root 'users#index'
+  get 'users', to: 'users#index'
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- **[OPTIONAL]** We can define these routes more concisely using the `resources` helper method.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```